w3af/core/ui/api/resources/sessions.py

```diff
@@ -0,0 +1,343 @@
+           methods=['GET'])
+@requires_auth
+@check_session_exists
+@check_plugin_type_exists
+@check_plugin_exists
+def get_plugin_config(**kwargs):
+    """
+    Shows current configuration and help text for the named plugin.
+
+    :return: 404 if scan or plugin does not exist.
+             Otherwise, a JSON object containing settings and basic help text.
+    """
+    scan_id = kwargs['scan_id']
+    plugin_type = kwargs['plugin_type']
+    plugin = kwargs['plugin']
+
+    w3af = SCANS[scan_id].w3af_core
+
+    opts = (
+        w3af.plugins.get_plugin_options(plugin_type, plugin) or
+        w3af.plugins.get_plugin_inst(plugin_type, plugin).get_options()
+        )
+    plugin_opts = { i.get_name():
+            { "value": str(i.get_value()),
+              "description": i.get_desc(),
+              "type": i.get_type(),
+              "default": str(i.get_default_value()) }
+        for i in opts }
+
+    long_desc = " ".join(w3af.plugins.get_quick_instance(plugin_type, plugin)
+                         .get_long_desc().split())
+    enabled = plugin in w3af.plugins.get_enabled_plugins(plugin_type)
+
+    return jsonify({ 'configuration': plugin_opts,
+                     'description': long_desc,
+                     'enabled' : enabled })
+
+
+@app.route('/sessions/<int:scan_id>/plugins/<string:plugin_type>/<string:plugin>/',
+           methods=['PATCH'])
+@requires_auth
+@check_session_exists
+@check_plugin_type_exists
+@check_plugin_exists
+def set_plugin_config(**kwargs):
+    """
+    Allows a user to modify a single element of plugin configuration by sending
+    a PATCH request.
+
+    :return: 404 if scan or plugin does not exist.
+             400 if the PATCH request is not in valid format.
+             422 if the PATCH request is in valid format, but settings cannot be
+             applied (eg, negative numbers where none are accepted).
+
+             Otherwise, a JSON object confirming success and echoing the
+             changed setting.
+    """
+
+    scan_id = kwargs['scan_id']
+    plugin_type = kwargs['plugin_type']
+    plugin = kwargs['plugin']
+
+    w3af = SCANS[scan_id].w3af_core
+
+    plugin_opts = (
+        w3af.plugins.get_plugin_options(plugin_type, plugin) or
+        w3af.plugins.get_plugin_inst(plugin_type, plugin).get_options()
+    )
+
+    opt_names = dict(request.get_json(force=True))
+
+    if ('enabled' in opt_names and len(opt_names) == 1):
+        with SET_OPTIONS_LOCK:
+            enable_or_disable_plugin(w3af,
+                                     plugin,
+                                     plugin_type,
+                                     enable=request.json['enabled'])
+            return jsonify({
+                'message': 'success',
+                'modified': request.json
+                })
+    try:
+        # Remove 'enabled' if included as we must set this separately
+        opt_names.pop('enabled')
```

Comment:
Good catch, thanks. I'm afraid I've taken the more restrictive path and caught this in the 'add' method for `OptionList` - while the namespace method seems more elegant, it seems as though it would be confusing to allow 'enabled' as a plugin option.